### PR TITLE
Quadlinear

### PR DESCRIPTION
**Pull request by SSmaal**
Added two classes to be used for fatigue analysis according to British standards in stead of DNV

### CODE REVIEW

File: fatpack/endurance.py [hunk 1]
```diff
@@ -411,3 +506,187 @@ def get_stress(self, N):
...
+    the quadlinear curve. 
+
+    Example
+    -------
+    >>> import fatpack
+    >>> import numpy as np
+    >>> np.random.seed(10)
+
+    First we create an endurance curve with detail category 90.
+    >>> curve = fatpack.BiLinearEnduranceCurve(90.)
```
Comment (bdvandenberg) [inline fatpack/endurance.py:565]: I believe this has not been customized for Quad yet?

File: fatpack/endurance.py [hunk 2]
```diff
@@ -411,3 +506,187 @@ def get_stress(self, N):
...
+
+    Let us find the damage according to Miner's linear damage rule from
+    a rainflow counted signal y
+
+    >>> y = np.random.normal(size=100000) * 10.
+    >>> S = fatpack.find_rainflow_ranges(y)
+    >>> D = curve.find_miner_sum(S)
```
Comment (bdvandenberg) [inline fatpack/endurance.py:572]: Have you tested if this function works as well?

File: fatpack/endurance.py [hunk 3]
```diff
@@ -411,3 +506,187 @@ def get_stress(self, N):
...
+    >>> print("The damage in signal y is D={0:3.2e}".format(D))
+    The damage in signal y is D=1.19e-04
```
Comment (bdvandenberg) [inline fatpack/endurance.py:574]: Please update the example such that the result can be checked with the code (so you should be able to reproduce the outcome).

File: fatpack/check_SN_curve.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+plt.xlabel("Endurance")
+plt.ylabel("Stress range (MPa)")
+plt.grid(which='both')
+
+
+fatigue_limit = curve.get_stress(10**7)
+print("Fatigue limit:", fatigue_limit)
+
+loga2 = np.log10(curve.get_endurance(1.0))
+print("Intercept of second linear curve log a2:", loga2)
```
Comment (bdvandenberg) [inline fatpack/check_SN_curve.py:55]: Please move this code to an example jupyter notebook in fatpack/examples.

File: fatpack/endurance.py [hunk 5]
```diff
@@ -411,3 +506,187 @@ def get_stress(self, N):
...
+     s   Sk +- - - - - - - - - - - - - - - - - - - *
+     s      |    .    .                 .          . *      m4
+            |    .    .                 .          .    * - - - -+
+            |    .    .                 .          .       *     . 1
+            |    .    .                 .          .          *  .
+            |    .    .                 .          .             *
+            |    .    .                 .          .                *
+            |    .    .                 .          .                   *                                
+            |----|----|-----------------|----------|---------------------->
+                 Nc   Nd                Nl         Nk
```
Comment (bdvandenberg) [inline fatpack/endurance.py:549]: Nice plotting 😄 🥇

**Review by bdvandenberg — CHANGES_REQUESTED** [review]
Nice work Stijn! Just a few comments, nothing big.

File: fatpack/endurance.py (hunk 2 above)
Comment (SSmaal) [reply fatpack/endurance.py:572]: did now

File: fatpack/check_SN_curve.py (hunk 4 above)
Comment (SSmaal) [reply fatpack/check_SN_curve.py:55]: Ill just delete this, this example is already there

**Review by bdvandenberg — APPROVED** [review]

**Pull request merged**